### PR TITLE
🚀 feat(PlanCard): Make ListItem clickable

### PR DESCRIPTION
**Pull request by gabrielmaschke**
[JIRA Issue](https://gympass.atlassian.net/browse/EUEG-2847)

## Description 📄

- Add prop `onClick` to `PlanCard.ListItem` component and transform the wrapper in a button to make it accessible
- Change proptype of the prop `text`, it should accept components such as `Skeleton`

```jsx
<PlanCard discount="SAVE 40%">
  <PlanCard.Tag variant="informative">Active</PlanCard.Tag>
  <PlanCard.Content
    title="Gold"
    currency="$"
    price="99.90"
    period="/mo"
    badgeColor="energy"
  >
    <PlanCard.List>
      <PlanCard.ListItem icon={Checkin} text="2.900 gyms and studios" onClick={() => foo()} />
      <PlanCard.ListItem icon={Dumbbell} text="1x/day standard access" onClick={() => foo()} />
      <PlanCard.ListItem icon={User} text="4x/month 1-on-1 sessions" onClick={() => foo()} />
    </PlanCard.List>
  </PlanCard.Content>
</PlanCard>
```

## Platforms 📲

- [x] Web
- [ ] Mobile

## Type of change 🔍

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested? 🧪

[Enter the tips to test this PR]

- [x] Unit Test
- [x] Snapshot Test

## Checklist: 🔍

- [x] My code follows the contribution guide of this project [Contributing Guide](https://github.com/Gympass/yoga/blob/master/CONTRIBUTING.md)
- [ ] Layout matches design prototype: [FIGMA](https://figma.com/file/YOUR_LINK)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

## Screenshots 📸

No visual changes.

### CODE REVIEW

**Review by leonardokl — APPROVED** [review]

**Review by CaioAugustoR — APPROVED** [review]

**Review by luanlorenzo — APPROVED** [review]

**Review by alycecristines — APPROVED** [review]

File: packages/yoga/src/Card/web/PlanCard/List.jsx [hunk 1]
```diff
@@ -31,11 +32,20 @@ const Item = styled.li`
   }
 `;
 
-const Wrapper = styled.div`
+const Wrapper = styled(Box)`
   display: flex;
   min-width: 0;
   align-items: center;
 
+  ${props =>
```
Comment (MicaelRodrigues) [inline packages/yoga/src/Card/web/PlanCard/List.jsx:40]: Clicks can also be a link, Should we add that option? Example:
If the component receives a `href` prop, we can assume element should be a `<a>` tag..

WDYT?

Comment (MicaelRodrigues) [reply packages/yoga/src/Card/web/PlanCard/List.jsx:40]: Also, beware that `button` should have the "type" attribute. (Maybe we can assume always `type="button"`).

A helpful article: https://www.builder.io/blog/buttons

Comment (gabrielmaschke) [reply packages/yoga/src/Card/web/PlanCard/List.jsx:40]: > Clicks can also be a link, Should we add that option? Example: If the component receives a `href` prop, we can assume element should be a `<a>` tag..
> 
> WDYT?

I have discussed with the team, and for now, links are out of scope, at least within the context of Gympass.

Comment (gabrielmaschke) [reply packages/yoga/src/Card/web/PlanCard/List.jsx:40]: > Also, beware that `button` should have the "type" attribute. (Maybe we can assume always `type="button"`).
> 
> A helpful article: https://www.builder.io/blog/buttons

Nice article! I completely agree. I added the button type in this commit: 5845523425325a78f71ac90ce709fe97d41866d7.


@MicaelRodrigues could you review it again please?

**Review by matheushdsbr — APPROVED** [review]

**Pull request merged**